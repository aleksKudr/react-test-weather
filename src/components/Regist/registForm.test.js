import React from 'react';
import { Field } from 'redux-form';


it('test form', () => {
  const wrapper = shallow(
    <form
      onSubmit={jest.fn()}
      className="ui form error"
    />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('test Field User', () => {
  const wrapper = shallow(
    <Field
      name="username"
      component={jest.fn()}
      label="Login"
    />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('test Field Password', () => {
  const wrapper = shallow(
    <Field
      name="password"
      component={jest.fn()}
      label="Password"
    />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});
it('test label', () => {
  const wrapper = shallow(
    <label>
      {jest.fn()}
    </label>
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('test input', () => {
  const wrapper = shallow(
    <input
      {...jest.fn()}
      autoComplete="off"
    />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('test input', () => {
  const wrapper = shallow(
    <button className="ui button primary">Submit</button>
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});
