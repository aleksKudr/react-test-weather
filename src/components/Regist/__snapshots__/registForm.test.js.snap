// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Field Password 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    component={[MockFunction]}
    label="Password"
    name="password"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "component": [MockFunction],
      "forwardedRef": null,
      "label": "Password",
      "name": "password",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "component": [MockFunction],
        "forwardedRef": null,
        "label": "Password",
        "name": "password",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test Field User 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    component={[MockFunction]}
    label="Login"
    name="username"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "component": [MockFunction],
      "forwardedRef": null,
      "label": "Login",
      "name": "username",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "component": [MockFunction],
        "forwardedRef": null,
        "label": "Login",
        "name": "username",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <form
    className="ui form error"
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "ui form error",
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "ui form error",
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <input
    _isMockFunction={true}
    autoComplete="off"
    getMockImplementation={[Function]}
    getMockName={[Function]}
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": [MockFunction],
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": [MockFunction],
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test input 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <button
    className="ui button primary"
  >
    Submit
  </button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Submit",
      "className": "ui button primary",
    },
    "ref": null,
    "rendered": "Submit",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Submit",
        "className": "ui button primary",
      },
      "ref": null,
      "rendered": "Submit",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <label>
    [MockFunction]
  </label>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": [MockFunction],
    },
    "ref": null,
    "rendered": [MockFunction],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": [MockFunction],
      },
      "ref": null,
      "rendered": [MockFunction],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
